### ISSUE TITLE
Rule idea: warn against interpolating expressions with a unit

### ISSUE THREAD

**Issue by stof** [stylelint-scss/stylelint-scss#1122]
```scss
$var: #{$some-expr}px
```

A common mistake is to interpolate an expression followed by a unit and expecting this produce a number with that unit. However, this actually produces an unquoted string (which is then incompatible with any math function for instance). This is so common than it deserved a `heads up` note in the first section of the documentation about interpolation: https://sass-lang.com/documentation/interpolation/#in-sass-script

I would suggest a lint rule when an interpolation is followed by a known unit.
